src/js/components/TableControls.js

```diff
@@ -102,46 +116,61 @@ function TableControls({ cameraName, allColNames, selected, setSelected }) {
     columnCount: numControlColumns,
   }
 
-  const panelClass = !controlsOpen ? "table-panel" : "table-panel open"
+  const panelContainerClass = !controlsOpen
+    ? "table-controls-container"
+    : "table-controls-container open"
 
-  return (
-    <>
-      <div className={panelClass}>
-        <button
-          className="table-control-button"
-          onClick={toggleControls}
-          onKeyDown={handleKeyDown}
-          title="Add/Remove Columns"
-          aria-label="Add/Remove Columns"
-          aria-expanded={controlsOpen}
-          aria-controls="table-controls"
-          aria-haspopup="true"
-        >
-          Add/Remove Columns
-        </button>
+  const renderCheckbox = (title) => {
+    const isAvailable = availableColumns.includes(title)
+    const isSelected = selected.includes(title)
+
+    return (
+      <div
+        className={`table-option${!isAvailable ? " unavailable" : ""}`}
+        key={title}
+      >
+        <label htmlFor={title}>
+          <input
+            type="checkbox"
+            id={title}
+            name={title}
+            value={1}
+            checked={isSelected}
+            onChange={() => handleCheckboxChange(title)}
+            onKeyDown={handleKeyDown}
+            disabled={!isAvailable}
+          />
+          {title}
+        </label>
+      </div>
+    )
+  }
 
+  return (
+    <div className={panelContainerClass} id="table-controls">
+      <button
+        className="table-control-button"
+        onClick={toggleControls}
+        onKeyDown={handleKeyDown}
+        title="Add/Remove Columns"
+        aria-label="Add/Remove Columns"
+        aria-expanded={controlsOpen}
+        aria-controls="table-controls"
+        aria-haspopup="true"
+      >
+        Add/Remove Columns
+      </button>
+      <div className="table-panel">
         {controlsOpen && (
-          <div className="table-controls" style={gridStyle}>
-            {allColNames.map((title) => (
-              <div className="table-control" key={title}>
-                <label htmlFor={title}>
-                  <input
-                    type="checkbox"
-                    id={title}
-                    name={title}
-                    value={1}
-                    checked={selected.includes(title)}
-                    onChange={() => handleCheckboxChange(title)}
-                    onKeyDown={handleKeyDown}
-                  />
-                  {title}
-                </label>
-              </div>
-            ))}
+          <div className="table-options" style={gridStyle}>
+            {/* Show both available and unavailable selected columns */}
+            {Array.from(new Set([...selected, ...allColNames]))
+              .toSorted((a, b) => a.localeCompare(b))
+              .map(renderCheckbox)}
           </div>
         )}
       </div>
-    </>
+    </div>
   )
 }
 TableControls.propTypes = {
```

Comment:
You should add `availableColumns` and remove `metadata` and `isHistorical` from the `TableControls.propTypes`.